### PR TITLE
tensor.To() to copy the Tensor memory to a specify device

### PR DESCRIPTION
**Pull request by Yancey1989**
related issued #123 
Users can specify a Tensor device like:

``` golang
device := torch.NewDevice("cuda")
x := torch.RandN([]int64{2,3}).To(device)
...
```

### CODE REVIEW

File: tensor.go [hunk 1]
```diff
@@ -283,3 +291,8 @@ func View(a Tensor, shape []int64) Tensor {
 	SetTensorFinalizer((*unsafe.Pointer)(&t))
 	return Tensor{(*unsafe.Pointer)(&t)}
 }
+
+// To copy Tensor memory to the specify device
+func To(a Tensor, device Device) Tensor {
```
Comment (wangkuiyi) [inline tensor.go:296]: Just curious: does PyTorch have such a global function?

File: tensor.go [hunk 2]
```diff
@@ -218,6 +218,14 @@ func (a Tensor) Grad() Tensor {
 	return Tensor{(*unsafe.Pointer)(&t)}
 }
 
+// To returns a Tensor with the specify device
+func (a Tensor) To(device Device) Tensor {
+	var t C.Tensor
+	MustNil(unsafe.Pointer(C.Tensor_To(C.Tensor(*a.T), device.T, &t)))
```
Comment (wangkuiyi) [inline tensor.go:224]: It seems that we need to enrich the comment:

```go
// To returns a Tensor on the specified device with the same content as the a.
// If the specified device doesn't exist, To panics.
```

File: device_test.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	a := assert.New(t)
+	var device torch.Device
+	if torch.IsCUDAAvailable() {
+		device = torch.NewDevice("cuda")
+	} else {
+		device = torch.NewDevice("cpu")
+	}
+	a.NotNil(device)
+	x := torch.RandN([]int64{2, 3}, false)
+	x.To(device)
```
Comment (wangkuiyi) [inline device_test.go:20]: Should we add [`assert.NotPanics`](https://godoc.org/github.com/stretchr/testify/assert#NotPanics) check here?

**Comment by wangkuiyi** [comment]
Facebook releases CUDA and CPU-only libtorch; do we need to edit our `Makefile.linux`, or to add a new Makefile, to download and use the CUDA version?

**Comment by wangkuiyi** [comment]
@Yancey1989 I added `cgotorch/Makefile.linux-gpu` in https://github.com/Yancey1989/gotorch/pull/1. Please review.

File: tensor.go (hunk 1 above)
Comment (Yancey1989) [reply tensor.go:296]: Yes, c.f. https://github.com/pytorch/pytorch/blob/3544f60f7602081398ee62bc5d652a87f4743dab/aten/src/ATen/native/TensorConversions.cpp#L77

**Pull request merged**